src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -1812,8 +1812,10 @@ internal void AssignNameSlots()
                 _capnumlist = new Int32[_capcount];
                 int i = 0;
 
-                for (IDictionaryEnumerator de = _caps.GetEnumerator(); de.MoveNext();)
-                    _capnumlist[i++] = (int)de.Key;
+                foreach (KeyValuePair<int, int> kvp in _caps)
```

Comment:
This could iterate over `_caps.Keys`, but I'm not sure it makes any difference either way.